packages/network-controller/tests/provider-api-tests/block-hash-in-response.ts

```diff
@@ -367,305 +365,24 @@ export function testsForRpcMethodsThatCheckForBlockHashInResponse(
-                // specified except that the block param is replaced with
-                // the latest block number.
-                failoverComms.mockRpcCall({
-                  request,
-                  response: {
-                    result: 'ok',
-                  },
-                });
-
-                const messenger = buildRootMessenger();
-
-                const result = await withNetworkClient(
-                  {
-                    providerType,
-                    failoverRpcUrls: ['https://failover.endpoint'],
-                    messenger,
-                    getRpcServiceOptions: (rpcEndpointUrl) => {
-                      const commonOptions = { fetch, btoa };
-                      // We need to return different results.
-                      // eslint-disable-next-line jest/no-conditional-in-test
-                      if (rpcEndpointUrl === 'https://failover.endpoint') {
-                        const headers: HeadersInit = {
-                          'X-Baz': 'Qux',
-                        };
-                        return {
-                          ...commonOptions,
-                          fetchOptions: {
-                            headers,
-                          },
-                        };
-                      }
-                      const headers: HeadersInit = {
-                        'X-Foo': 'Bar',
-                      };
-                      return {
-                        ...commonOptions,
-                        fetchOptions: {
-                          headers,
-                        },
-                        policyOptions: {
-                          backoff: new ConstantBackoff(backoffDuration),
-                          maxRetries: 2,
-                          maxConsecutiveFailures: 6,
-                        },
-                      };
-                    },
-                  },
-                  async ({ makeRpcCall, clock }) => {
-                    messenger.subscribe(
-                      'NetworkController:rpcEndpointRequestRetried',
-                      () => {
-                        // Ensure that we advance to the next RPC request
-                        // retry, not the next block tracker request.
-                        // We also don't need to await this, it just needs to
-                        // be added to the promise queue.
-                        // eslint-disable-next-line @typescript-eslint/no-floating-promises
-                        clock.tickAsync(backoffDuration);
-                      },
-                    );
-
-                    for (let i = 0; i < 5; i++) {
-                      await ignoreRejection(makeRpcCall(request));
-                    }
-                    return await makeRpcCall(request);
-                  },
-                );
-
-                expect(result).toBe('ok');
-              },
-            );
-          },
-        );
+      testsForRpcFailoverBehavior({
```

Comment:
There are a large amount of changes in this file, but I refactored the tests for the RPC failover behavior out into their own file, `rpc-failover.ts`, as they were really getting difficult to manage. I will point out the new test I added there.